storage/leveldb/leveldb.go

```diff
@@ -151,11 +150,13 @@ func (s *DB) Get(key []byte) ([]byte, error) {
 		return nil, storage.ErrDBIsClosed
 	}
 
+	isRemoved := s.batch.IsRemoved(key)
+	if isRemoved {
+		return nil, storage.ErrKeyNotFound
+	}
+
 	data := s.batch.Get(key)
 	if data != nil {
```

Comment:
[question]

Couldn't we avoid this by making `Get` from `batch.go` return error and have ErrKeyNotFound when key is not found?

Reply:
could have been done like that. However I do not intend to change the batcher behavior just for this. The main reason: the implication it might arise.